src/lib/coda_base/user_command_status.ml

```diff
@@ -125,6 +144,537 @@ module Failure = struct
+
+    let receiver_not_present = {none with receiver_not_present= true}
+
+    let amount_insufficient_to_create_account =
+      {none with amount_insufficient_to_create_account= true}
+
+    let cannot_pay_creation_fee_in_token =
+      {none with cannot_pay_creation_fee_in_token= true}
+
+    let source_insufficient_balance =
+      {none with source_insufficient_balance= true}
+
+    let source_minimum_balance_violation =
+      {none with source_minimum_balance_violation= true}
+
+    let receiver_already_exists = {none with receiver_already_exists= true}
+
+    let not_token_owner = {none with not_token_owner= true}
+
+    let mismatched_token_permissions =
+      {none with mismatched_token_permissions= true}
+
+    let overflow = {none with overflow= true}
+
+    let signed_command_on_snapp_account =
+      {none with signed_command_on_snapp_account= true}
+
+    let snapp_account_not_present = {none with snapp_account_not_present= true}
+
+    let update_not_permitted = {none with update_not_permitted= true}
+
+    let incorrect_nonce = {none with incorrect_nonce= true}
+
+    let to_enum = function
+      | {predicate= true; _} ->
+          1
+      | {source_not_present= true; _} ->
+          2
+      | {receiver_not_present= true; _} ->
+          3
+      | {amount_insufficient_to_create_account= true; _} ->
+          4
+      | {cannot_pay_creation_fee_in_token= true; _} ->
+          5
+      | {source_insufficient_balance= true; _} ->
+          6
+      | {source_minimum_balance_violation= true; _} ->
+          7
+      | {receiver_already_exists= true; _} ->
+          8
+      | {not_token_owner= true; _} ->
+          9
+      | {mismatched_token_permissions= true; _} ->
+          10
+      | {overflow= true; _} ->
+          11
+      | {signed_command_on_snapp_account= true; _} ->
+          12
+      | {snapp_account_not_present= true; _} ->
+          13
+      | {update_not_permitted= true; _} ->
+          14
+      | {incorrect_nonce= true; _} ->
+          15
+      | _ ->
+          0
+
+    let of_enum = function
+      | 0 ->
+          Some none
+      | 1 ->
+          Some predicate
+      | 2 ->
+          Some source_not_present
+      | 3 ->
+          Some receiver_not_present
+      | 4 ->
+          Some amount_insufficient_to_create_account
+      | 5 ->
+          Some cannot_pay_creation_fee_in_token
+      | 6 ->
+          Some source_insufficient_balance
+      | 7 ->
+          Some source_minimum_balance_violation
+      | 8 ->
+          Some receiver_already_exists
+      | 9 ->
+          Some not_token_owner
+      | 10 ->
+          Some mismatched_token_permissions
+      | 11 ->
+          Some overflow
+      | 12 ->
+          Some signed_command_on_snapp_account
+      | 13 ->
+          Some snapp_account_not_present
+      | 14 ->
+          Some update_not_permitted
+      | 15 ->
+          Some incorrect_nonce
+      | _ ->
+          None
+
+    let min = 0
```

Comment:
could use `[%%define_from_scope (failure_min,failure_max)]`, if you take the above suggestion